### DISCUSSION TITLE
Execution time

### DISCUSSION THREAD

**Discussion by ghost** [b-inary/postflop-solver#24] [Q&A]
Hi, I'm not a programmer, so I don't know how much I'm doing wrong. But when I run "cargo run --release --example basic" with a configuration similar to the WASM Postflop web, my execution time is the same. Shouldn't it be lower and similar to the Desktop app? Is there something else that needs to be configured? I'm on Mac and I haven't tried the desktop app yet, but I assumed that the Rust code would be faster than the web.

**Comment by b-inary** [comment]
Apparently, the Wasm version is actually quite fast, as far as I can confirm on my M2 MacBook Air. Here is the current benchmark on an M2 MacBook Air under the same conditions as [this comparison](https://github.com/b-inary/wasm-postflop#comparison) (8 threads, 32-bit FP mode, 0.1% exploitability):

- WASM Postflop (Chrome 113): 46.7s
- Desktop Postflop: 35.7s

Note that both apps are compiled with the `--features custom-alloc` flag for performance; without this flag, the native performance would be about 3% worse.

**Reply by ghost** [reply]
Thanks. Is there any way to print the data of the entire tree raw? That is, the ev, strategy etc of each hand for each action, I want to get something similar to the json output of TexasSolver (OOP strategy; IP strategy for action 0; ...; OOP strategy for each possible turn...)

**Reply by b-inary** [reply]
Unfortunately, we do not provide an easy interface for such output, so please write your own code by referring [the documentation](https://b-inary.github.io/postflop_solver/postflop_solver/). If you need help, I may be able to help you with specific questions.
